docs/strategies/retry.md

```diff
@@ -108,6 +108,54 @@ new ResiliencePipelineBuilder<HttpResponseMessage>().AddRetry(optionsExtractDela
 | `OnRetry`          | `null`                                                                     | Action executed when retry occurs.                                                       |
 | `MaxDelay`         | `null`                                                                     | Caps the calculated retry delay to a specified maximum duration.                         |
 
+## Calculation of the next delay
+
+If the `ShouldHandle` predicate returns `true` and the next attempt number is not greater than `MaxRetryAttempts` then the retry strategy calculates the next delay.
+
+There are many properties that may contribute to this calculation:
+
+- `BackoffType`: It specifies which calculation algorithm should run.
+- `Delay`: If only this one is specified then it will be used as is. If others are defined as well then this will be used as a *base delay*.
+- `DelayGenerator`: If specified then it will overwrite other properties based calculation. **Except** if it returns a negative `TimeSpan` where the other properties based calculation is used.
+- `MaxDelay`: If specified then it caps the delay if the calculated one is greater than this value. **Except** if `DelayGenerator` is used where no capping is applied.
```

Comment:
```suggestion
- `MaxDelay`: If specified then it caps the delay if the calculated one is greater than this value. **Except** if `DelayGenerator` is used where no capping is applied.
```

Just wondering, shall we apply the capping to values produced by generator too?

Reply:
I would say yes. That was my intuition before I checked the implementation. But let's deal with that in a separate PR if you don't mind.

Reply:
My intuition says yes too, the less special cases, the better.

Reply:
I would agree we should apply it - `Retry-After` isn't necessarily authoritative and the user may wish to try earlier by overriding. If we do that we should make it clear in the documentation.

Reply:
It is a concision decision by the strategy creator that (s)he sets the `MaxDelay`. If it is set then the strategy should cap the value regardless how it is calculated/retrieved IMHO.